### PR TITLE
Eve-7 Additional checks and minor corrections

### PR DESCRIPTION
**Pull request by alja**
These changes are needed for use of CMS web event display.

### CODE REVIEW

**Comment by alja** [comment]
@linev Please check.

**Comment by linev** [comment]
@phsft-bot build

File: graf3d/eve7/inc/ROOT/REveManager.hxx [hunk 1]
```diff
@@ -217,6 +217,8 @@ public:
 
    void EnforceTimerActive (Bool_t ta) { fTimerActive = ta; }
 
+   RWebWindow* GetWebWindow() const { return fWebWindow.get(); }
+
```
Comment (linev) [inline graf3d/eve7/inc/ROOT/REveManager.hxx:221]: I recommend to use here `std::shared_ptr<RWebWindow>` as return value

File: graf3d/eve7/inc/ROOT/REveDataCollection.hxx [hunk 2]
```diff
@@ -101,6 +92,12 @@ public:
 
    using REveElement::GetHighlightTooltip;
    std::string GetHighlightTooltip(const std::set<int>& secondary_idcs) const override;
+
+   void SetItemsChangeDelegate (std::function<void (REveDataItemList*, const std::vector<int>&)> handler_func);
+   void SetFillImpliedSelectedDelegate (std::function<void (REveDataItemList*, Set_t& impSelSet)> handler_func);
```
Comment (linev) [inline graf3d/eve7/inc/ROOT/REveDataCollection.hxx:97]: 1. Please use typedefs for function types.
2. Use these typedefs also when declaring class members `_handler_items_change` and `_handler_fillimp`
3. Use default naming convention for these members like `fHandlerItemsChange`

File: graf3d/eve7/src/REveDataCollection.cxx [hunk 3]
```diff
@@ -79,28 +88,30 @@ void REveDataItemList::ItemChanged(Int_t idx)
...
    std::vector<int> ids;
    ids.push_back(idx);
-   _handler_items_change(this, ids);
+   _handler_items_change( this , ids);
 }
 
 //______________________________________________________________________________
 
-void REveDataItemList::FillImpliedSelectedSet(Set_t &impSelSet)
+void REveDataItemList::FillImpliedSelectedSet( Set_t& impSelSet)
```
Comment (linev) [inline graf3d/eve7/src/REveDataCollection.cxx:96]: According to clang-format it should be `Set_t &impSelSet` - please use it like this

File: graf3d/eve7/src/REveDataCollection.cxx [hunk 4]
```diff
@@ -118,52 +129,53 @@ Bool_t REveDataItemList::SetRnrState(Bool_t iRnrSelf)
...
 }
 
 //______________________________________________________________________________
-void REveDataItemList::ProcessSelection(ElementId_t selectionId, bool multi, bool secondary,
-                                        const std::set<int> &secondary_idcs)
+void REveDataItemList::ProcessSelection(ElementId_t selectionId, bool multi, bool secondary, const std::set<int>& secondary_idcs)
 {
    RefSelectedSet() = secondary_idcs;
-   REveSelection *selection = (REveSelection *)ROOT::Experimental::gEve->FindElementById(selectionId);
+   REveSelection* selection = (REveSelection*) ROOT::Experimental::gEve->FindElementById(selectionId);
```
Comment (linev) [inline graf3d/eve7/src/REveDataCollection.cxx:148]: Same clang-format problem here.
Also when doing explicit cast, one should better use `static_cast<REveSelection *>()`

File: graf3d/eve7/src/REveDataProxyBuilderBase.cxx [hunk 5]
```diff
@@ -75,6 +75,10 @@ void REveDataProxyBuilderBase::Build()
          auto itemSize = m_collection->GetNItems(); //cashed
 
          Clean();
+
+         if (!m_collection->GetRnrSelf())
+         return;
```
Comment (linev) [inline graf3d/eve7/src/REveDataProxyBuilderBase.cxx:80]: Please use correct indentation:
```
if (!m_collection->GetRnrSelf())
   return;
```

**Review by linev — CHANGES_REQUESTED** [review]

**Comment by alja** [comment]
@lineve Thank you for the corrections!

**Comment by linev** [comment]
@alja, can you provide and use typedef for functions:
```
 std::function<void (REveDataItemList*, const std::vector<int>&)> 
 std::function<void (REveDataItemList*, Set_t& impSel)>
```

**Comment by alja** [comment]
@linev
I don't have an alias for a plain vector of integer which is used in ItemsChange callback. Alias Set_t is std:set of EveElement pointers and is used in FillImpliedSelected callback through REveSelection mechanism.
I agree Set_t can be confused with integer set. It is extensively used n eve and I would not change it right now.

**Comment by linev** [comment]
I mean use 
```
typedef  std::function<void (REveDataItemList*, const std::vector<int>&)> ItemListVectorFunc_t;  
```
And then use ` ItemListVectorFunc_t` as type of data member and as type of setter argument

**Comment by linev** [comment]
@phsft-bot build

**Review by linev — APPROVED** [review]

**Comment by alja** [comment]
@linev The alias for the functions made code more readable. Thank You!

**Pull request merged**